packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,301 @@
+}
+
+/**
+ * A CloudFront distribution with associated origin(s) and caching behavior(s).
+ *
+ * @experimental
+ */
+export class Distribution extends Resource implements IDistribution {
+
+  /**
+   * Creates a Distribution construct that represents an external (imported) distribution.
+   */
+  public static fromDistributionAttributes(scope: Construct, id: string, attrs: DistributionAttributes): IDistribution {
+    return new class extends Resource implements IDistribution {
+      public readonly domainName: string;
+      public readonly distributionId: string;
+
+      constructor() {
+        super(scope, id);
+        this.domainName = attrs.domainName;
+        this.distributionId = attrs.distributionId;
+      }
+    }();
+  }
+
+  /**
+   * Creates a Distribution for an S3 Bucket, where the bucket has not been configured for website hosting.
+   *
+   * This creates a single-origin distribution with a single default behavior.
+   */
+  public static forBucket(scope: Construct, id: string, bucket: s3.IBucket): Distribution {
+    return new Distribution(scope, id, {
+      defaultBehavior: { origin: Origin.fromBucket(scope, 'SingleOriginBucket', bucket) },
+    });
+  }
+
+  /**
+   * Creates a Distribution for an S3 bucket, where the bucket has been configured for website hosting.
+   *
+   * This creates a single-origin distribution with a single default behavior.
+   */
+  public static forWebsiteBucket(scope: Construct, id: string, bucket: s3.IBucket): Distribution {
+    return new Distribution(scope, id, {
+      defaultBehavior: { origin: Origin.fromWebsiteBucket(scope, 'SingleOriginWebsiteBucket', bucket) },
+    });
+  }
+
+  public readonly domainName: string;
+  public readonly distributionId: string;
+
+  private readonly defaultBehavior: CacheBehavior;
+  private readonly additionalBehaviors: CacheBehavior[] = [];
+
+  private readonly errorConfigurations: CfnDistribution.CustomErrorResponseProperty[];
+  private readonly certificate?: acm.ICertificate;
+
+  constructor(scope: Construct, id: string, props: DistributionProps) {
+    super(scope, id);
+
+    if (props.certificate) {
+      const certificateRegion = Stack.of(this).parseArn(props.certificate.certificateArn).region;
+      if (!Token.isUnresolved(certificateRegion) && certificateRegion !== 'us-east-1') {
+        throw new Error('Distribution certificates must be in the us-east-1 region.');
+      }
+    }
+
+    this.defaultBehavior = new CacheBehavior({ pathPattern: '*', ...props.defaultBehavior });
+    if (props.additionalBehaviors) {
+      Object.entries(props.additionalBehaviors).forEach(([pathPattern, behaviorOptions]) => {
+        this.addBehavior(pathPattern, behaviorOptions);
+      });
+    }
+
+    this.certificate = props.certificate;
+    this.errorConfigurations = props.errorConfigurations ?? [];
+
+    const distribution = new CfnDistribution(this, 'CFDistribution', { distributionConfig: {
+      enabled: true,
```

Comment:
Feels like this should be configurable, I can a scenario to temporarily disable a distribution...

Reply:
Fair. I was planning on adding it to the DistributionProps in the next wave -- where I'm aiming for more coverage/completeness. For this initial commit, I figured this was a sane thing to default on.